sematic/resolvers/state_machine_resolver.py

```diff
@@ -183,6 +185,72 @@ def _run_inline(self, future: AbstractFuture) -> None:
     def _wait_for_scheduled_runs(self) -> None:
         pass
 
+    def _fail_future_with_timeout(
+        self,
+        executing_future: AbstractFuture,
+        timeout_restricting_future: typing.Optional[AbstractFuture] = None,
+    ):
+        if timeout_restricting_future is None:
+            timeout_restricting_future = executing_future
+
+        if timeout_restricting_future.id == executing_future.id:
+            message = (
+                f"The Sematic function {executing_future.calculator.__name__} did not "
+                f"complete in time to satisfy its "
+                f"{executing_future.props.timeout_minutes} minute timeout."
+            )
+        else:
+            message = (
+                f"The Sematic function {executing_future.calculator.__name__} did not "
+                f"complete in time to satisfy the "
+                f"{timeout_restricting_future.props.timeout_minutes} minute timeout "
+                f"specified by its parent "
+                f"{timeout_restricting_future.calculator.__name__}."
+            )
+        self._handle_future_failure(
+            executing_future,
+            format_exception_for_run(TimeoutError(message)),
+        )
+
+    def _wait_for_scheduled_runs_with_timeout(self) -> None:
+        seconds_until_next_timeout, future = self._get_seconds_to_next_timeout()
+        if seconds_until_next_timeout is not None and future is not None:
+            logger.info(
```

Comment:
This log will print a negative number before failing, which is not the best experience. Please move it after the if.

Reply:
done